homeassistant/components/dhcp/__init__.py

```diff
@@ -248,10 +248,10 @@ def _async_process_device_state(self, state: State):
             return
 
         ip_address = attributes.get(ATTR_IP)
-        hostname = attributes.get(ATTR_HOST_NAME)
+        hostname = attributes.get(ATTR_HOST_NAME) or ""
```

Comment:
```suggestion
        hostname = attributes.get(ATTR_HOST_NAME, "")
```

Reply:
Good call. Sometimes I can't see the forest through the trees. 🙃